File: 1787-sum-of-absolute-differences-in-a-sorted-array/sum-of-absolute-differences-in-a-sorted-array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        total_sum = sum(nums)
        left_sum = 0
        output = [0 for _ in range(len(nums))]

        for idx, num in enumerate(nums):
            right_sum = total_sum - num - left_sum
            left_size = idx
            right_size = len(nums) - idx - 1

            output[idx] = ((left_size * num) - left_sum) + (right_sum - (right_size * num))
            left_sum += num

        return output
        
        

File: 1787-sum-of-absolute-differences-in-a-sorted-array/sum-of-absolute-differences-in-a-sorted-array.js
var getSumAbsoluteDifferences = function(nums) {
    let totalSum = nums.reduce((a, b)=> a + b, 0)
    let leftSum = 0
    let currentSum = 0
    let output = new Array(nums.length).fill(0)

    for (let i = 0; i < nums.length; i++) {
        let leftSize = i
        rightSize = nums.length - i - 1
        let rightSum = totalSum - leftSum - nums[i]
        output[i] = ((leftSize * nums[i]) - leftSum) + (rightSum - (rightSize * nums[i]))
        leftSum += nums[i]
    }

    return output
};